lib/ruby_smb/client.rb

```diff
@@ -138,6 +138,11 @@ class Client
     #   @return [Boolean]
     attr_accessor :smb1
 
+    # Challenge value returned by the server during SMB1 negotiation with Extended Security turned off
+    # @!attribute [rw] challenge
+    #       @return [String]
+    attr_accessor :challenge
```

Comment:
Is `challenge` something that is only possible for SMB1 with Extended Security turned off? I have the feeling that `challenge` also exists for SMB1 with Ext. Sec, SMB2 and SMB3. If it is, it might be a good idea to add support for every case scenario and give access to the `challenge` in any situation.